.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install shellcheck
+        run: |
+          sudo apt-get update && sudo apt-get install -y shellcheck
+
+      - name: Run shellcheck
+        run: |
+          shellcheck wcurl tests/*
+
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install shunit2
+        run: |
+          if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+            sudo apt-get update
+            sudo apt-get install -y shunit2
+          elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            brew install shunit2
+          fi
+
+      - name: Run shunit2 tests
+        run: |
+          shunit2 ./tests/*
```

Comment:
The correct way to call the tests is `./tests/tests.sh`, I see that they are failing in this PR, maybe this is the reason?

Reply:
The reason for those tests to fail is because the PR https://github.com/curl/wcurl/pull/31 needs to be merged first.

On that code change the **tests.sh** and **README.md** had been update to work when the command `shunit2 ./tests/*` is executed. This also open doors for splitting or organize tests across multiple shell file.

Reply:
If we can keep calling the tests with the current approach, I would suggest we don't change it in this CI PR.

There are side effects that we should discuss for a change like that, such as breakage for users that already run the tests with the current way, or inability to add non-test scripts to the `tests/*` folder. These might be all fine things to do, but it needs to be discussed.

If we end up deciding it's ok to change the way the tests are invoked, this PR should also change the tests as ultimately they need to pass in this PR, to show that it works.

Reply:
> I would suggest we don't change it in this CI PR.

We need to rely on this change in the PR otherwise `shellcheck` step will fail:

```
In tests/tests.sh line 204:
. shunit2
  ^-----^ SC1091 (info): Not following: shunit2 was not specified as input (see shellcheck -x).
  ```
  
  Another option it to ignore `SC1091 (info)` for that line and keep using `tests/tests.sh`

Reply:
Let's discuss this further to see what is the best approach.

1. Run the test as is  running it using `./tests/tests.sh` and skip some info message `SC1091` when running `shellcheck`
2. Change the way we execute the tests and leave `shellcheck` with no overrides.

Reply:
Create an issue to track silent test failures because of `shunit2` version.

https://github.com/curl/wcurl/issues/39